contentcuration/contentcuration/frontend/shared/views/MarkdownEditor/MarkdownEditor/MarkdownEditor.vue

```diff
@@ -487,20 +487,50 @@
 
         // Prevent Squire from deleting custom editor nodes when the cursor is left of one.
         const isCustomNode = node => node && node.hasAttribute && node.hasAttribute('is');
+
         const getElementAtRelativeOffset = (selection, offset) =>
           selection &&
           squire.getSelectionInfoByOffset(selection.endContainer, selection.endOffset + offset)
             .element;
+
         const getLeftwardElement = selection => getElementAtRelativeOffset(selection, -1);
         const getRightwardElement = selection => getElementAtRelativeOffset(selection, 1);
+
+        const getCharacterAtRelativeOffset = (selection, relativeOffset) => {
+          let { element, offset } = squire.getSelectionInfoByOffset(
+            selection.startContainer,
+            selection.startOffset + relativeOffset
+          );
+          return element.nodeType === document.TEXT_NODE && element.textContent[offset];
+        };
+
+        const spacerAndCustomElementAreLeftward = selection =>
+          selection &&
+          isCustomNode(getElementAtRelativeOffset(selection, -2)) &&
+          selection.startContainer.nodeType === document.TEXT_NODE &&
+          getCharacterAtRelativeOffset(selection, -1) &&
+          !!getCharacterAtRelativeOffset(selection, -1).match(/^\s/);
+
+        const spacerAndCustomElementAreRightward = selection =>
+          selection &&
+          isCustomNode(getElementAtRelativeOffset(selection, 2)) &&
+          selection.startContainer.nodeType === document.TEXT_NODE &&
+          getCharacterAtRelativeOffset(selection, 0) &&
+          !!getCharacterAtRelativeOffset(selection, 0).match(/\s$/);
```

Comment:
Since you're just getting the Boolean, it'd be more performant to use `RegExp.test()` like:

`/\s$/.test(getCharacterAtRelativeOffset(selection, 0))` 

here and at line 512 above. 

Then you don't have to `!!` what you get back from `String.match()`

Reply:
The main benefit of `String.match()` if I recall correctly is that you can get the values of matching groups back in an array if you have groups in your regex.